### DISCUSSION TITLE
Introducing FluentCLI: a rust based command line interface for Langflow, (and competitors)

### DISCUSSION THREAD

**Discussion by njfio** [langflow-ai/langflow#1962] [Show and tell]
Hi! 

For the past six weeks, I've been building a Rust cli application to interact with Langflow. In full disclosure, it started with Flowise, and I added Langflow afterward. With the latest release (1.0.0a36) of Langflow, I'm very excited to explore Langflow's capabilities further and complete the rest of the FluentCLI functionality with it.  

[Loom Demo](https://www.loom.com/share/c65b2a01b94243358db6540f3d8f4841)

This is a list of functionality that FluentCLI provides.

- Multi-Engine Support: Interact seamlessly with FlowiseAI, Langflow, and Webhook workflows.
- Simplified Request Handling: Send requests to your workflows with a simple command structure.
- Context Management: Provide additional context via <stdin> or files for richer interactions.
- Configuration Management: Load and modify workflow configurations from a centralized JSON file.
- Environment Variable Integration: A secure method to store sensitive information, such as API keys, in environment variables and reference them in your configurations.
- Inline Override: Easily override configuration inline for greater control over your workflows.
- File Upload: Upload images and other files to your workflows. (Untested to date with Langflow)
- Upsert Functionality: Send JSON payloads and upload files to endpoints for data management. (partially working with Langflow)
Output Customization: You can control the output format, including stylized markdown, parsed code blocks, and full JSON responses.
- Media Download: Download media files directly linked to CLI responses.
- Autocomplete Generation: Generate Bash, Powershell, and Fig autocomplete scripts for enhanced usability.
- Amber Secure Value Integration: Securely decrypt and manage sensitive keys using the Amber secrets management tool.

[FluentCLI Repository](https://github.com/njfio/fluent_cli)

FluentCLI provides unique capabilities and allows your Langflow AI workflows to easily be injected into existing processes.

**Comment by Cristhianzl** [comment]
Hi! 👋

We are using other channels as our official means of communication with users. We apologize for the delayed response.
For faster assistance, please direct your questions to our Discord channel or open a GitHub issue, as we are much more active on these platforms.

[Langflow Discord Server](https://discord.com/invite/EqksyE2EX9)

Thank you for your understanding.

Best regards,
The Langflow Team
